validator/sawtooth_validator/gossip/gossip.py

```diff
@@ -204,9 +204,11 @@ def peer_to_public_key(self, peer):
     def get_peers_public_keys(self):
         """Returns the list of public keys for all peers."""
         with self._lock:
-            return [
-                self._network.connection_id_to_public_key(peer)
-                for peer in copy.copy(self._peers)]
+            keys = [self._network.connection_id_to_public_key(peer)
```

Comment:
You could do this as a generator expression within the list comprehension so that you are not iterating through the list twice (once for the list comprehension and once for the filter). May come at the cost of readability though.

Reply:
Done